lib/resty/healthcheck.lua

```diff
@@ -582,11 +598,13 @@ end
 -- and returns `true`.
 -- @param ip IP address of the target being checked.
 -- @param port the port being checked against.
+-- @param hostname (optional) hostname of the target being checked.
 -- @param http_status the http statuscode, or nil to report an invalid http response.
 -- @param check (optional) the type of check, either "passive" or "active", default "passive".
 -- @return `true` on success, `nil` if the status was ignored (not in active or
 -- passive health check lists) or `nil + error` on failure.
-function checker:report_http_status(ip, port, http_status, check)
+function checker:report_http_status(ip, port, hostname, http_status, check)
+  hostname = hostname or ip
```

Comment:
Let's move those fallbacks (there are several of them) in `key_for` instead, which would be a lot more robust. It would also save some space in those strings when we don't have a hostname instead of duplicating it.

```lua
local function key_for(key_prefix, ip, port, hostname)
  return string.format("%s:%s:%s%s", key_prefix, ip, port, hostname and ":" .. hostname or "")
end
```

Reply:
Done!